engine/src/flutter/shell/platform/android/io/flutter/plugin/view/SensitiveContentPlugin.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.view;
+
+import android.app.Activity;
+import android.view.View;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.systemchannels.SensitiveContentChannel;
+import io.flutter.plugin.common.MethodChannel;
+
+/**
+ * {@link SensitiveContentPlugin} is the implementation of all functionality needed to set
+ * the content sensitivity level of a native Flutter Android {@code View}.
+ *
+ * <p>This plugin handles requests for setting content sensitivity sent by the {@link
+ * io.flutter.embedding.engine.systemchannels.SensitiveContentChannel}.
+ */
+public class SensitiveContentPlugin
+    implements SensitiveContentChannel.SensitiveContentMethodHandler {
+
+  private final Activity mflutterActivity;
+  private final SensitiveContentChannel mSensitiveContentChannel;
+
+  public SensitiveContentPlugin(
+      @NonNull Activity flutterActivity, @NonNull SensitiveContentChannel sensitiveContentChannel) {
+    mflutterActivity = flutterActivity;
+    mSensitiveContentChannel = sensitiveContentChannel;
+
+    mSensitiveContentChannel.setSensitiveContentMethodHandler(this);
+  }
+
+  /**
+   * Sets content sensitivity level of the Android {@code View} with the specified {@code
+   * flutterViewId} to the level specified by {@contentSensitivity}.
+   */
+  @Override
+  public void setContentSensitivity(
+      @NonNull int flutterViewId,
+      @NonNull int contentSensitivity,
+      @NonNull MethodChannel.Result result) {
+    final View flutterView = mflutterActivity.findViewById(flutterViewId);
+    if (flutterView == null) {
+      result.error("error", "Requested Flutter View to set content sensitivty of not found.", null);
+    }
+
+    // Set contentSensitivity on the requested View.
+    flutterView.setContentSensitivity(contentSensitivity);
+
+    // Invalidate the View to force a redraw if we require that the screen
+    // become unobscured, which is the case where the View was previously
+    // marked sensitive but now is no longer.
+    final int currentContentSensitivity = flutterView.getContentSensitivity();
+    final boolean shouldInvalidateView =
+        currentContentSensitivity == View.CONTENT_SENSITIVITY_SENSITIVE
+            && contentSensitivity != View.CONTENT_SENSITIVITY_SENSITIVE;
```

Comment:
I found this a bit tough to parse but what I am seeing is that if the view is sensitive and we just set the view to sensitive the invalidate.

I think the logic you want is 
```
final int currentContentSensitivity = flutterView.getContentSensitivity();
if (currentContentSensitivity != contentSensitivity) { 
    flutterView.setContentSensitivity(contentSensitivity);
    flutterView.invalidate();
}
```
This code is easier to read and only invalidates the view on a changed state. 
Also think the passed in value should be requestedContentSensitivity or something that indicates how it is different from current.

Reply:
Changed naming, but kept my logic because we only have to invalidate if we are changing away from sensitive state.

Reply:
In general I think when the state changes in any direction we should do the same thing. Selective view invalidation can cause hard to reproduce bugs that are very difficult to reason about.

Reply:
Also I would still encourage you to more clearly distinguish between current and requested but you dont have to use those prefixes.